subsys/bluetooth/mesh/sensor.c

```diff
@@ -61,21 +61,19 @@ bool bt_mesh_sensor_delta_threshold(const struct bt_mesh_sensor *sensor,
 		thrsh_mill = (prev_mill * thrsh_mill) / (100LL * 1000000LL);
 	}
 
-	if (BT_DBG_ENABLED) {
-		char delta_str[BT_MESH_SENSOR_CH_STR_LEN];
-		char curr_str[BT_MESH_SENSOR_CH_STR_LEN];
-		char prev_str[BT_MESH_SENSOR_CH_STR_LEN];
-		char thrsh_str[BT_MESH_SENSOR_CH_STR_LEN];
+	char delta_str[BT_MESH_SENSOR_CH_STR_LEN];
```

Comment:
Is it optimized-out when debug logging is disabled? https://godbolt.org/z/qbczqn4hY suggests it may not. Maybe it's worth adding a condition on LOG_LEVEL?